src/DocumentDB.ChangeFeedProcessor/ChangeFeedProcessorBuilder.cs

```diff
@@ -473,15 +474,26 @@ private IPartitionManager BuildPartitionManager(ILeaseStoreManager leaseStoreMan
                     collection.PartitionKey != null &&
                     collection.PartitionKey.Paths != null &&
                     collection.PartitionKey.Paths.Count > 0;
-                if (isPartitioned && isPartitionKeyByIdRequiredIfPartitioned &&
-                    (collection.PartitionKey.Paths.Count != 1 || collection.PartitionKey.Paths[0] != "/id"))
+
+                if (isPartitioned &&
+                    (collection.PartitionKey.Paths.Count != 1 ||
+                    (collection.PartitionKey.Paths[0] != IdPkPathName && collection.PartitionKey.Paths[0] != PartitionKeyPkPathName)))
                 {
-                    throw new ArgumentException("The lease collection, if partitioned, must have partition key equal to id.");
+                    throw new ArgumentException($"The lease collection, if partitioned, must have partition key equal to {IdPkPathName} or {PartitionKeyPkPathName}.");
                 }
 
-                var requestOptionsFactory = isPartitioned ?
-                    (IRequestOptionsFactory)new PartitionedByIdCollectionRequestOptionsFactory() :
-                    (IRequestOptionsFactory)new SinglePartitionRequestOptionsFactory();
+                IRequestOptionsFactory requestOptionsFactory = null;
+                if (isPartitioned)
+                {
+                    // we allow only id or leasePk partitioning so check only flag
+                    requestOptionsFactory = collection.PartitionKey.Paths[0] != PartitionKeyPkPathName ?
```

Comment:
nit: better to check for == rather than !=. This will avoid extra churn in future.